OrbitGl/App.cpp

```diff
@@ -814,39 +777,89 @@ void OrbitApp::LoadModules(const std::shared_ptr<Process>& process, int32_t proc
     }
     modules_currently_loading_.insert(module->m_FullName);
 
-    // TODO (159889010) Move symbol loading off the main thread.
-    const std::string& module_path = module->m_FullName;
-    const std::string& build_id = module->m_DebugSignature;
-    if (build_id.empty()) {
-      LOG("Warning: Module \"%s\" does not contain a build id. Orbit cannot use local symbol files "
-          "or cache symbols without a build id.",
-          module_path);
-      LoadModuleOnRemote(process, process_id, module, preset);
+    const auto& symbols_path = FindSymbolsLocally(module->m_FullName, module->m_DebugSignature);
+    if (symbols_path) {
+      LoadSymbols(symbols_path.value(), process, module, preset);
       continue;
     }
 
-    auto scoped_status = CreateScopedStatus(
-        absl::StrFormat("Trying to find symbols for \"%s\" on local machine...", module_path));
-    auto symbols = symbol_helper_.LoadUsingSymbolsPathFile(module_path, build_id);
+    LoadModuleOnRemote(process, module, preset);
+  }
+}
 
-    // Try loading from the cache
-    if (!symbols) {
-      const std::string cached_file_name = symbol_helper_.GenerateCachedFileName(module_path);
-      symbols = symbol_helper_.LoadSymbolsFromFile(cached_file_name, build_id);
+ErrorMessageOr<std::filesystem::path> OrbitApp::FindSymbolsLocally(
+    const std::filesystem::path& module_path, const std::string& build_id) {
+  const auto scoped_status = CreateScopedStatus(absl::StrFormat(
+      "Searching for symbols on local machine (module: \"%s\")...", module_path.string()));
+
+  if (build_id.empty()) {
+    return ErrorMessage(absl::StrFormat(
+        "Unable to find local symbols for module \"%s\", build id is empty", module_path.string()));
+  }
+
+  std::string error_message;
+  {
+    const auto symbols_path = symbol_helper_.FindSymbolsWithSymbolsPathFile(module_path, build_id);
+    if (symbols_path) {
+      LOG("Found symbols for module \"%s\" in user provided symbol folder. symbols filename: "
+          "\"%s\"",
+          module_path.string(), symbols_path.value().string());
+      return symbols_path.value();
     }
+    error_message += "\n* " + symbols_path.error().message();
+  }
+  {
+    const auto symbols_path = symbol_helper_.FindSymbolsInCache(module_path, build_id);
+    if (symbols_path) {
+      LOG("Found symbols for module \"%s\" in cache. symbols filename: \"%s\"",
+          module_path.string(), symbols_path.value().string());
+      return symbols_path.value();
+    }
+    error_message += "\n* " + symbols_path.error().message();
+  }
+  error_message =
+      absl::StrFormat("Did not find local symbols for module \"%s\":", module_path.string()) +
```

Comment:
```suggestion
      absl::StrFormat("Did not find local symbols for module \"%s\": ", module_path.string()) +
```

Reply:
also is there a reason for not using `: %s, .., error_message`?

Reply:
No, there is not. I dont know why I did this. -> done